homework-submission/barbarosapp/src/App.js

```diff
@@ -0,0 +1,62 @@
+import React, { Component } from "react";
+import "./App.css";
+import logo from "./logo.svg";
+import ItemList from "./components/itemList";
+import Data from "./data/data";
+import AddMenu from "./components/addMenu";
+
+class App extends Component {
+  state = { Data };
+
+  handleCheck = id => {
+    const newData = this.state.Data.map(item => {
+      if (item.id === id) {
+        return { ...item, done: !item.done };
+      }
+      return item;
+    });
+
+    this.setState({ Data: newData });
+  };
+
+  handleAdd = (name, deadline) => {
+    const newId = this.state.Data.length;
+    const addedData = this.state.Data;
+    const addition = {
+      id: newId,
+      name,
+      deadline,
+      done: false
+    };
+    this.setState({ Data: [...addedData, addition] });
+  };
+
+  handleRemove = item => {
+    const newData = this.state.Data;
+    newData.splice(item.id, 1);
+    this.setState({ Data: newData });
```

Comment:
Two things are here:
 1. you are mutating the state directly with splice.
 2. you may use `.filter` instead of splice making better performance for react while `.filter` is returning a new array, it means you have a complete new state.

Reply:
Corrected by using filter